specifications/transactions/offchain-api/README.md

```diff
@@ -0,0 +1,256 @@
+
+`invalid_jws`: invalid JWS format (compact) or protected header
+
+`invalid_jws_signature`: JWS signature verification failed
+
+####### Request Object Validation Error Codes
+
+`invalid_json`: decoded JWS body is not json.
+
+`invalid_object`: command request/response object json is not object, or the command object type does not match `command_type`.
+
+`missing_field`: A required field contains no data or is missing.
+
+`unknown_field`: field is unknown for an object.
+
+`unknown_command_type`: invalid/unsupported `command_type`.
+
+`invalid_field_value`: A field does not match the expected type, e.g., a UUID is not correctly formatted or an enum field contains an invalid value.
+
+`invalid_recipient_signature`:
+* Field `recipient_signature` value is not hex-encoded bytes.
+* Field `recipient_signature` value is an invalid signature.
+
+## Network Error Handling
+
+In the case of network failure, the sending party for a Command is expected to re-send the Command until it gets a response from the counterparty VASP. An exponential backoff is suggested for Command re-sends.
+
+Upon receipt of a Command that has already been processed (resulting in a success response or a Command error), the receiving side must reply with the same response as was previously issued (successful commands, or commands that fail with a command error, are idempotent). Requests that resulted in protocol errors may result in different responses.
+
+For example:
+* VASP A sends a request to VASP B which failed with an unknown network error. VASP A retries the send request to VASP B.
+* VASP B receives both requests, and processes them in parallel.
+* Both requests attempt to acquire a common lock. The first request that VASP A believes failed acquires the lock and the second request fails.
+* VASP B successfully processes the first request and sends back a successful response, however, VASP A doesn’t receive the response due to the previous errors.
+* VASP B rejects the second request due to lock contention.
+* VASP A receives the failure for the second request, and later, retries the same request. If VASP B processes the request when the lock is not under contention it will respond successfully.
+
+## JWS scheme {#jws-scheme}
+
+All `CommandRequestObject` and `CommandResponseObject` messages exchanged on the off-chain channel between two services must be signed using a specific configuration of the JWS scheme.
+
+The JSON Web Signature (JWS) scheme is specified in RFC 7515. Messages in the off-chain channel are signed with a specific configuration:
+
+* The JWS Signature Scheme used is EdDSA as specified in RFC 8032 (EdDSA) and RFC 8037 (Elliptic Curve signatures for JWS).
+* The JWS Serialization scheme used is Compact as specified in [Section 3.1 of RFC 7515](https://tools.ietf.org/html/rfc7515#section-3.1) 
+* The Protected Header should contain the JSON object {"alg": "EdDSA"}, indicating the signature algorithm used.
+* The Unprotected header must be empty
```

Comment:
`Compact` encoding does not support unprotected headers.

Reply:
This is pulled from our DIP :)